x-pack/plugins/cases/server/client/configure/validators.test.ts

```diff
@@ -69,4 +76,417 @@ describe('validators', () => {
       ).not.toThrow();
     });
   });
+
+  describe('validateTemplateKeysAgainstConfiguration', () => {
+    beforeEach(() => {
+      jest.clearAllMocks();
+    });
+
+    it('does not throw if all templates are in configuration', () => {
+      expect(() =>
+        validateTemplateKeysAgainstConfiguration({
+          requestTemplateFields: [
+            {
+              key: 'template_key_1',
+              name: 'first template',
+              description: 'this is a first template value',
+              caseFields: null,
+            },
+            {
+              key: 'template_key_2',
+              name: 'second template',
+              description: 'this is a second template value',
+              caseFields: {
+                title: 'Case title with template 2',
+              },
+            },
+          ],
+          templatesConfiguration: [
+            {
+              key: 'template_key_1',
+              name: 'first template',
+              description: 'this is a first template value',
+              caseFields: {
+                tags: ['first-template'],
+              },
+            },
+            {
+              key: 'template_key_2',
+              name: 'second template',
+              description: 'this is a second template value',
+              caseFields: null,
+            },
+          ],
+        })
+      ).not.toThrow();
+    });
+
+    it('does not throw if no custom fields are in request', () => {
```

Comment:
Is this title correct? `validateTemplateKeysAgainstConfiguration` doesn't deal with custom fields.

Reply:
moved to `validateTemplatesCustomFieldsInRequest ` in https://github.com/elastic/kibana/pull/182575/commits/a7b6a1b1525615fb9bbd815b19e18cfbb802eee8